### DISCUSSION TITLE
Return entities of a Team in the same order the entities were added

### DISCUSSION THREAD

**Discussion by riazufila** [PaperMC/Paper#11314] [Ideas]
### Is your feature request related to a problem?

It'd be a good thing to know which Entity joined first and which Entity joined last to form some sort of hierarchy in a Team.

### Describe the solution you'd like.

Return a LinkedHashSet<String> instead of Set<String> such as one in Team#getEntries. Or, if maintaining order for the whole Entities in a Team is not possible, it'd be nice to just have a getter for the very first Entity that joined the team.

### Describe alternatives you've considered.

Persist the datetime which Entity joined a Team. Having this feature implemented would remove requirement to persist such detail.

**Comment by Machine-Maker** [comment]
I would just "wrap" a bukkit team in your own type that tracked that information. The order of entries is not defined. So what I'd do is save your "teams" somewhere, load them and create a bukkit team from it, but then wrap it in your own type that tracks that information and only make changes through that wrapper type. Then you can save that back to whatever storage you want.

**Reply by riazufila** [reply]
Thanks! Seems like I'm already doing almost the same thing after I wrote this request. Will go ahead and close!
